data/provider/src/main/java/tech/pegasys/teku/api/stateselector/StateSelectorFactory.java

```diff
@@ -145,11 +149,13 @@ public StateSelector forStateRoot(final Bytes32 stateRoot) {
             .thenApply(
                 maybeState ->
                     maybeState.map(
-                        state ->
-                            addMetaData(
-                                state,
-                                client.isOptimisticBlock(
-                                    BeaconBlockHeader.fromState(state).getRoot()))));
+                        state -> {
+                          BeaconBlockHeader header = BeaconBlockHeader.fromState(state);
+                          return addMetaData(
+                              state,
+                              client.isOptimisticBlock(header.getRoot()),
+                              client.isCanonicalBlock(header.getSlot(), header.getRoot()));
```

Comment:
This should use `state.getSlot()` as the block header might be from an earlier slot where the block is canonical but the state isn't because the state is from an empty slot and the canonical chain has a block present.